src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -637,14 +718,18 @@ static void replace_allocation(PhaseIterGVN* igvn, Node* res, Node* dom) {
 
 Node* InlineTypeNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   Node* oop = get_oop();
-  if (is_default(phase) && inline_klass()->is_initialized() &&
+  // An InlineTypeNode in larval state is up for updation and
+  // should not be replaced by precomputed default oops.
+  Node* alloc = AllocateNode::Ideal_allocation(oop, phase);
+  bool is_larval_alloc = alloc && alloc->as_Allocate()->_larval == true;
+  if (!is_larval_alloc && is_default(phase) && inline_klass()->is_initialized() &&
```

Comment:
The similar change is also merged into the valhalla mainline. See: https://github.com/openjdk/valhalla/pull/828.